### PR TITLE
CrashingRunner: cleanup some code

### PR DESCRIPTION
**Pull request by dhalperi**
make it final, fix an error message, remove unused code

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @tgroh

**Comment by tgroh** [comment]
LGTM
(LVGTM)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CrashingRunner.java [hunk 1]
```diff
@@ -41,32 +39,11 @@ public PipelineResult run(Pipeline pipeline) {
     throw new IllegalArgumentException(String.format("Cannot call #run(Pipeline) on an instance "
             + "of %s. %s should only be used as the default to construct a Pipeline "
             + "using %s, and cannot execute Pipelines. Instead, specify a %s "
-            + "by providing PipelineOptions in the environment variable '%s'.",
-        getClass().getSimpleName(),
-        getClass().getSimpleName(),
+            + "by providing PipelineOptions in the system property '%s'.",
+        CrashingRunner.class.getSimpleName(),
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CrashingRunner.java]: Why this? The former is (ever so slightly) more robust, no?

**Comment by kennknowles** [comment]
LGTM from committer. One nit, not a big deal.

**Comment by dhalperi** [comment]
It's more readable. The class is final, and if we ever change the name via the automated refactoring will get it. Didn't see much benefit, but also not much harm and it made the code epsilon more readable IMO

**Pull request closed without merging**